File: mycollab-services/src/main/java/com/esofthead/mycollab/module/ecm/service/ExternalResourceService.java
/**
 * This file is part of mycollab-services.
 *
 * mycollab-services is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * mycollab-services is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with mycollab-services.  If not, see <http://www.gnu.org/licenses/>.
 */
package com.esofthead.mycollab.module.ecm.service;

import java.io.InputStream;
import java.util.List;

import com.esofthead.mycollab.cache.IgnoreCacheClass;
import com.esofthead.mycollab.core.persistence.service.IService;
import com.esofthead.mycollab.module.ecm.domain.Content;
import com.esofthead.mycollab.module.ecm.domain.ExternalDrive;
import com.esofthead.mycollab.module.ecm.domain.ExternalFolder;
import com.esofthead.mycollab.module.ecm.domain.Folder;
import com.esofthead.mycollab.module.ecm.domain.Resource;

/**
 * 
 * @author MyCollab Ltd.
 * @since 1.0
 * 
 */
@IgnoreCacheClass
public interface ExternalResourceService extends IService {
	/**
	 * 
	 * @param drive
	 * @param path
	 * @return
	 */
	List<Resource> getResources(ExternalDrive drive, String path);

	/**
	 * 
	 * @param drive
	 * @param path
	 * @return
	 */
	List<ExternalFolder> getSubFolders(ExternalDrive drive, String path);

	/**
	 * 
	 * @param drive
	 * @param path
	 * @return
	 */
	Resource getCurrentResourceByPath(ExternalDrive drive, String path);

	/**
	 * 
	 * @param drive
	 * @param childPath
	 * @return
	 */
	Folder getParentResourceFolder(ExternalDrive drive, String childPath);

	/**
	 * 
	 * @param drive
	 * @param path
	 * @return
	 */
	Folder createFolder(ExternalDrive drive, String path);

	/**
	 * 
	 * @param drive
	 * @param content
	 * @param in
	 */
	void saveContent(ExternalDrive drive, Content content, InputStream in);

	/**
	 * 
	 * @param drive
	 * @param oldPath
	 * @param newPath
	 */
	void rename(ExternalDrive drive, String oldPath, String newPath);

	/**
	 * 
	 * @param drive
	 * @param path
	 */
	void deleteResource(ExternalDrive drive, String path);

	/**
	 * 
	 * @param drive
	 * @param path
	 * @return
	 */
	InputStream download(ExternalDrive drive, String path);

	/**
	 * 
	 * @param drive
	 * @param fromPath
	 * @param toPath
	 */
	void move(ExternalDrive drive, String fromPath, String toPath);
}

File: mycollab-services/src/main/java/com/mycollab/module/ecm/service/ExternalResourceService.kt
/**
 * Copyright © MyCollab
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Affero General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU Affero General Public License for more details.
 *
 * You should have received a copy of the GNU Affero General Public License
 * along with this program.  If not, see <http://www.gnu.org/licenses/>.
 */
package com.mycollab.module.ecm.service

import com.mycollab.cache.IgnoreCacheClass
import com.mycollab.db.persistence.service.IService
import com.mycollab.module.ecm.domain.*

import java.io.InputStream

/**
 * @author MyCollab Ltd.
 * @since 1.0
 */
@IgnoreCacheClass
interface ExternalResourceService : IService {
    /**
     * @param drive
     * @param path
     * @return
     */
    fun getResources(drive: ExternalDrive, path: String): List<Resource>

    /**
     * @param drive
     * @param path
     * @return
     */
    fun getSubFolders(drive: ExternalDrive, path: String): List<ExternalFolder>

    /**
     * @param drive
     * @param path
     * @return
     */
    fun getCurrentResourceByPath(drive: ExternalDrive, path: String): Resource

    /**
     * @param drive
     * @param childPath
     * @return
     */
    fun getParentResourceFolder(drive: ExternalDrive, childPath: String): Folder

    /**
     * @param drive
     * @param path
     * @return
     */
    fun createNewFolder(drive: ExternalDrive, path: String): Folder

    /**
     * @param drive
     * @param content
     * @param in
     */
    fun saveContent(drive: ExternalDrive, content: Content, `in`: InputStream)

    /**
     * @param drive
     * @param oldPath
     * @param newPath
     */
    fun rename(drive: ExternalDrive, oldPath: String, newPath: String)

    /**
     * @param drive
     * @param path
     */
    fun deleteResource(drive: ExternalDrive, path: String)

    /**
     * @param drive
     * @param path
     * @return
     */
    fun download(drive: ExternalDrive, path: String): InputStream

    /**
     * @param drive
     * @param fromPath
     * @param toPath
     */
    fun move(drive: ExternalDrive, fromPath: String, toPath: String)
}
